src/utils/merge-timeseries-data.ts

```diff
@@ -0,0 +1,51 @@
+type timeSeriesData = number[][];
+
+const lerp = (a: number, b: number, t: number) => a * (1 - t) + b * t;
+
+const lerpForTime = (array: timeSeriesData, time: number) => {
+  const firstTime =array[0][0];
+  const finalTime = array[array.length - 1][0];
+  if(time < firstTime) {
+    return array[0][1];
+  }
+  if(time > finalTime) {
+    return array[array.length - 1][1];
+  }
+  let lastValue = 0;
+  let lastTime = 0;
+  let index = 0
+  while(array[index][0] < time) {
+    lastValue = array[index][1];
+    lastTime = array[index][0];
+    index++;
+  }
+  const nextValue = array[index][1];
+  const nextTime = array[index][0];
+  if (nextTime === time) {
+    return nextValue;
+  }
+  const t = (time - lastTime) / (nextTime - lastTime);
+  return lerp(lastValue, nextValue, t);
+}
+
+// Given two input arrays of time series data, merge them into a single array
+// where each element is a triplet of  numbers representing a time value of a and b.
+export const mergeTimeSeriesData = (dataA: timeSeriesData, dataB:timeSeriesData) => {
```

Comment:
I think to be defensive you should check the length of both `dataA` and `dataB` before you start indexing into them

Reply:
Fixed in [a58a392](https://github.com/concord-consortium/sensor-interactive/pull/80/commits/a58a3927a52fc291015f8ef63f01613772e190a6)